### DISCUSSION TITLE
Introspect a schema

### DISCUSSION THREAD

**Discussion by pihentagy** [colinhacks/zod#1351] [Q&A]
What is the proper way to introspect a schema?

Given a zod type:

```ts
export const LovEntity = z.object({
  id: z.number(),
  code: z.string(),
  description: z.string(),
});
```

Now how can I provide some information, that: "There are 3 fields in this schema. id is number, code and description are strings."

Or to ask it in a different way: how can I iterate through the keys of an object and figure out their names and types?

thanks

**Comment by smakinson** [comment]
@pihentagy I'm just beginning to look for something like this as well. I am interested in getting the paths of a schema via: https://deepdash.io/#paths-keysdeep

Did you have any success?

**Comment by scotttrinh** [comment]
`ZodObject` has a `shape` property. This isn't well documented, but you can look at the ecosystem packages that do transformations from Zod and get of sense of how you might use it for your use case.

**Reply by smakinson** [reply]
I'll take a look at those, thank you for the reply.
